libraries/AP_GPS/AP_GPS.h

```diff
@@ -176,6 +176,8 @@ class AP_GPS
         return status(primary_instance);
     }
 
+    bool is_healthy(bool record_log, bool send_feedback);
```

Comment:
is_healthy should simply return if it's healthy or not, any other decisions (logging/mavlink) should all be done by the caller.